scripts/mysql/upgrade-0.5.0-to-0.5.1-mysql.sql

```diff
@@ -0,0 +1,25 @@
+--
+-- Copyright 2024 Datastrato Pvt Ltd.
+-- This software is licensed under the Apache License version 2.
+--
+ALTER TABLE `role_meta` MODIFY COLUMN `securable_object_full_name` VARCHAR(256) NOT NULL DEFAULT '' COMMENT 'deprecated';
+ALTER TABLE `role_meta` MODIFY COLUMN `securable_object_type` VARCHAR(32) NOT NULL DEFAULT '' COMMENT 'deprecated';
+ALTER TABLE `role_meta` MODIFY COLUMN `privileges`  VARCHAR(64) NOT NULL DEFAULT '' COMMENT 'deprecated';
+ALTER TABLE `role_meta` MODIFY COLUMN `privilege_conditions` VARCHAR(64) NOT NULL DEFAULT '' COMMENT 'deprecated';
+
+CREATE TABLE IF NOT EXISTS `role_meta_securable_object` (
+    `id` BIGINT(20) UNSIGNED NOT NULL COMMENT 'securable object id',
+    `role_id` BIGINT(20) UNSIGNED NOT NULL COMMENT 'role id',
+    `full_name` VARCHAR(246) NOT NULL COMMENT 'securable object full name',
+    `type`  VARCHAR(128) NOT NULL COMMENT 'securable object type',
+    `privilege_names` VARCHAR(256) NOT NULL COMMENT 'securable object privilege names',
+    `privilege_conditions` VARCHAR(256) NOT NULL COMMENT 'securable object privilege conditions',
+    `current_version` INT UNSIGNED NOT NULL DEFAULT 1 COMMENT 'securable objectcurrent version',
+    `last_version` INT UNSIGNED NOT NULL DEFAULT 1 COMMENT 'securable object last version',
+    `deleted_at` BIGINT(20) UNSIGNED NOT NULL DEFAULT 0 COMMENT 'securable object deleted at',
+    PRIMARY KEY (`id`),
+    KEY `idx_obj_rid` (`role_id`)
+    ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin COMMENT 'securable object meta';
+
+INSERT INTO `role_meta_securable_object` (`role_id`, `full_name`, `type`, `privilege_names`, `privilege_conditions`, `current_version`, `last_version`,  `deleted_at`)
+SELECT `role_id`, `securable_object_full_name`, `securable_object_type`, `privileges`, `privilege_conditions`, `current_version`, `last_version`,  `deleted_at`  from `role_meta`;
```

Comment:
@xloya can you please help to review this part?

Reply:
I think some judgment should be made here to avoid overwriting some old data. According to the current development logic, the following steps will be taken when users upgrade:
1. Create a new `role_meta_securable_object` table
2. Upgrade the server. After the upgrade is completed, new data will be written to the `role_meta_securable_object` table.
3. At this time, the `role_meta` table data needs to be flushed to `role_meta_securable_object`. This may cause new data to be overwritten by old data, so we should always give priority to `role_meta_securable_object` data. If data exists in `role_meta_securable_object`, there is no need to refresh. @qqqttt123 What do you think?